src/languageservice/services/yamlSchemaService.ts

```diff
@@ -282,6 +293,28 @@ export class YAMLSchemaService extends JSONSchemaService {
             return resolveSchema();
         }
     }
+    
+    /**
+     * Retrieve schema if declared as modeline
+     * @param doc 
+     */
+    private getSchemaFromModeline(doc: any) : string{
+        if (doc instanceof SingleYAMLDocument) {
+            const modelineDeclaration = '# yaml-language-server:';
+            const yamlLanguageServerModeline = doc.lineComments.find(lineComment => lineComment.startsWith(modelineDeclaration));
+            if (yamlLanguageServerModeline != undefined) {
+                const schemaKey = '$schema=';
+                const indexOfJsonSchemaParameter = yamlLanguageServerModeline.indexOf(schemaKey);
+                if (yamlLanguageServerModeline.indexOf(schemaKey) != -1) {
```

Comment:
You can eliminate the second call to index if with `if (indexOfJsonSchemaParameter != -1)`

Reply:
good catch (still not acustomed to this language where there is no good refactoring tooling compared to Java)